### ISSUE TITLE
using v:page.resources.fal on translated pages

### ISSUE THREAD

**Issue by ThomasWaw** [FluidTYPO3/vhs#996]
Hi there,
I'm using vhs 2.4.0 with TYPO3 7.6.4.

I use v:page.resources.fal with a page uid to get the Image related to this page via FAL in the Page-Properties. So far so good.    

When I use it with a translated version of a page - but same pageUid - I get a wrong Image.  
I get the Fal-Image of that Page that has the uid of the translation (pages_language_overlay) of the actual page. 

Is this somehow understandabele? 

kind regards
Thomas

**Comment by torvitas** [comment]
For us this most often results in Exceptions because our page uids start somewhere at 10000 so there are no pages with the same uid like the overlay records.

We worked around it by extending the ViewHelper and making sure it uses the correct table.

**Comment by ThomasWaw** [comment]
Hi torvitas,

thanks for your replay!
I think the viewhelper uses the right table but the wrong uid to look up: it uses the uid of the language_overlay rather then the one of the original page.
Anyway - I couldn't figure out the possible problem in the viewhelpers code and helped myself with good old TypoScript and the very usefull f:cObject-Viewhelper :)

kind regards

**Comment by torvitas** [comment]
Hi ThomasWaw,

the problem in the ViewHelpers is that at some point it tries to get the record for the uid you provide. That's where it crashes for us, because there is no page with such low uids for us. (like I said earlier, our page uids start at 11000, while the overlay uids start at 1)

So we just check if its a overlay record and then change the table to look up the (page/page overlay) record to pages_language_overlay. We do that in the initialize method after calling parent.

Greetings

**Comment by bjo3rnf** [comment]
Hi @torvitas,

mind providing a pull request with your solution?

Cheers
Björn

**Comment by ogrosko** [comment]
Isn't is something common with my problem here: https://github.com/FluidTYPO3/vhs/issues/862#issuecomment-206975929

**Comment by bjo3rnf** [comment]
Looks like it, yes. Give me a moment please.

**Comment by grimmcreative** [comment]
This fix works perfect for me https://github.com/ogrosko/vhs/pull/1/commits/1e647530e932e49d8c7b57c1c7133062cd9143cf

**Comment by sebastianschrama** [comment]
So what is the solution here? This is very confusing for me. I use VHS 3.0.1 with TYPO3 7.6.10 and the problem already exists for me. I use v:page.resources.fal and for translations I get the wrong images.

**Comment by NamelessCoder** [comment]
@sebastianschrama The solution is merged into our `development` branch which means it will be included in the next TER release. If you require the solution before such a release is made then please install VHS from git and check out the `development` branch.

**Comment by sebastianschrama** [comment]
Okay, thank you!

**Comment by MrMooky** [comment]
We're currently using VHS 4.4.0 along with TYPO3 8.7.11. That's how we grab the resources from the media field:

```
<f:if condition="{page.media} >= 1">
    <f:then>
    <v:resource.record.fal table="pages" field="media" uid="{page.uid}" as="resources">
        <f:for each="{resources}" as="resource">
            <f:image src="{resource.id}" treatIdAsReference="1" width="1400" height="350c" class="heroimg" />
        </f:for>
    </v:resource.record.fal>
    </f:then>
    <f:else>
    <v:resource.record.fal table="pages" field="media" uid="{page.pid}" as="resources">
        <f:for each="{resources}" as="resource">
            <f:image src="{resource.id}" treatIdAsReference="1" width="1400" height="350c" class="heroimg" />
        </f:for>
    </v:resource.record.fal>
    </f:else>
</f:if>
```

The problem we're having a the same @ThomasWaw had, I guess. The images show perfectly for the default language but are not used for the translated pages.

**Comment by pixelplant** [comment]
@NamelessCoder Wondering if this issue been fixed? I still get the same behaviour in vhs 4.40 and TYPO3 8.7.12
The media for the default language shows properly. For the translated pages, if I have a media defined, the overlay image shows ok, but if I don't have an image then the one defined for the default image should be displayed, which is not the case, or am I missing something?
How can we tell it to use the media from the default language if no media is specified for a translated page?

**Comment by iammati** [comment]
Experiencing the same behaviour on TYPO3 v10.4.13 + VHS v6.0.4.
Making a custom ViewHelper as a workaround for now but consider this ticket as it isn't fixed yet.
